packages/eslint-plugin-edu-react/src/rules/no-useless-key.ts

```diff
@@ -24,9 +24,12 @@ const rule: TSESLint.RuleModule<keyof typeof failureMessages> = {
         const entireJsxElement = openingJsxElement?.parent;
         const parent = entireJsxElement?.parent;
 
-        // If the parent is another JSX element, as opposed to a return statement or an array or
-        // something, we definitely don't need the `key` attribute.
-        if (parent?.type === 'JSXElement') {
+        // If the parent is another JSX element or a variable/const, as opposed to an array or
+        // return statement, we don't need the `key` attribute.
+        if (
+          parent?.type === 'JSXElement' ||
+          parent?.type === 'VariableDeclarator'
```

Comment:
@ahuth Why is it okay for a `return` to have a key but not for a `const` that then gets returned?

Reply:
`return` for Array mapping:

```tsx
<ul>
  {things.map((thing) => {
    return <li key={thing.id}>{thing.name}</li>
  })}
</ul>
```